MeshLib/MeshEditing/FlipElements.h

```diff
@@ -0,0 +1,36 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef FLIPELEMENTS_H_
+#define FLIPELEMENTS_H_
+
+#include <vector>
+#include "MeshLib/Elements/Element.h"
+
+namespace MeshLib
+{
+class Mesh;
+class Node;
+
+/**
+ * Reverses the node order of a 1D / 2D element such that the direction 
+ * of a line changes and normals of 2D elements changes its sign.
+ */
+MeshLib::Element* flipElement(MeshLib::Element const& elem, std::vector<MeshLib::Node*> const& nodes);
```

Comment:
Why do you need the second argument? If my understanding from the documentation is correct, the sequence of nodes of the element is changed only, i.e. no new nodes should be necessary.